File: src/test/java/com/natu/ftax/transaction/TransactionTestUtils.java
package com.natu.ftax.transaction;

import com.natu.ftax.transaction.domain.SubmitTransactionCommand;
import com.natu.ftax.transaction.domain.Transaction;

import java.math.BigDecimal;
import java.time.Instant;

public class TransactionTestUtils {

    public static Transaction draftTransaction(String id) {
        return Transaction.create(id);
    }

    public static Transaction submittedTransaction(String id) {
        Transaction tx = draftTransaction(id);
        tx.submit(new SubmitTransactionCommand(
                id,
                "transfer",
                Instant.now(),
                "BTC",
                "ETH",
                "BTC",
                BigDecimal.valueOf(1.0),
                BigDecimal.valueOf(2.0),
                BigDecimal.valueOf(0.01),
                "ext-123"
        ));
        return tx;
    }

    public static Transaction submittedTransactionWithValue(String id,
            double amountIn,
            double amountOut,
            double amountFee,
            String tokenIn,
            String tokenOut,
            String tokenFee) {
        Transaction tx = draftTransaction(id);
        tx.submit(new SubmitTransactionCommand(
                id,
                "transfer",
                Instant.now(),
                tokenIn,
                tokenOut,
                tokenFee,
                BigDecimal.valueOf(amountIn),
                BigDecimal.valueOf(amountOut),
                BigDecimal.valueOf(amountFee),
                "ext-123"
        ));
        return tx;
    }
}

File: src/test/kotlin/com/natu/ftax/transaction/TransactionTestUtils.kt
package com.natu.ftax.transaction

import com.natu.ftax.transaction.domain.SubmitTransactionCommand
import com.natu.ftax.transaction.domain.Token
import com.natu.ftax.transaction.domain.Transaction
import com.natu.ftax.transaction.domain.TransactionType
import java.util.*

class TransactionTestUtils {

    companion object {

        @JvmStatic
        fun draftTransaction(id: String): Transaction {
            return Transaction.create(id)
        }

        @JvmStatic
        fun submittedTransaction(id: String): Transaction {
            val tx = draftTransaction(id)
            tx.submit(command = SubmitTransactionCommand(
                id = id,
                transactionType = TransactionType.TRANSFER,
                date = Date(),
                tokenIn = "BTC",
                tokenOut = "ETH",
                tokenFee = "BTC",
                amountIn = 1.0,
                amountOut = 2.0,
                amountFee = 0.01,
                externalId = "ext-123"
            ))

            return tx
        }

        @JvmStatic
        fun submittedTransactionWithValue(id: String,
                                            amountIn: Double,
                                            amountOut: Double,
                                            amountFee: Double,
                                            tokenIn: Token = "BTC",
                                            tokenOut: Token = "ETH",
                                            tokenFee: Token = "BTC"): Transaction {

            val tx = draftTransaction(id)
            tx.submit(command = SubmitTransactionCommand(
                id = id,
                transactionType = TransactionType.TRANSFER,
                date = Date(),
                tokenIn = tokenIn,
                tokenOut = tokenOut,
                tokenFee = tokenFee,
                amountIn = amountIn,
                amountOut = amountOut,
                amountFee = amountFee,
                externalId = "ext-123"
            ))

            return tx
        }

    }
}